Improvements to differential rotation doc (#5515)
sunpy/sunpy: docs/code_ref/coordinates/rotatedsunframe.rst

*********************************************

Normally, coordinates refer to a point in inertial space (relative to the barycenter of the solar system).
Transforming to a different observation time does not move the point itself, but if the coordinate frame origin and/or axis change with time, the coordinate representation is updated to account for this change.

> In solar physics an example of a frame that changes with time is the heliographic Stonyhurst (HGS) frame. Its origin moves with the center of the Sun, and its x-axis to follow the Sun-Earth line.

Reviewer: ```suggestion
In solar physics an example of a frame that changes with time is the `~sunpy.coordinates.frames.HeliographicStonyhurst` (HGS) frame.
Its origin moves with the center of the Sun, and its orientation rotates such that the longitude component of Earth is zero at any given time.
```
To be accurate, the HGS x-axis is not exactly aligned with the Sun-Earth line.

A coordinate in a HGS frame of reference transformed to a HGS frame defined a day later will have a different longitude::

>   >>> from astropy.coordinates import ICRS, SkyCoord

Reviewer: ```suggestion
  >>> from astropy.coordinates import SkyCoord
```

  >>> import astropy.units as u
  >>> from sunpy.coordinates import HeliographicStonyhurst

  >>> hgs_coord = SkyCoord(0*u.deg, 0*u.deg, radius=1*u.au, frame='heliographic_stonyhurst', obstime="2001-01-01")
  >>> new_frame = HeliographicStonyhurst(obstime="2001-01-02")
  >>> new_hgs_coord = hgs_coord.transform_to(new_frame)
  >>> hgs_coord.lon, new_hgs_coord.lon
  (<Longitude 0. deg>, <Longitude -1.01372559 deg>)

but when transformed to an inertial frame of reference we can see that these two coordinates refer to the same point in space::

>   >>> hgs_coord.transform_to('icrs')
>   <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
>        (101.79107615, 26.05004621, 0.99601156)>
>   >>> new_hgs_coord.transform_to('icrs')
>   <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
>         (101.79107615, 26.05004621, 0.99601156)>

Reviewer: ```suggestion
  >>> hgs_coord.transform_to('icrs')
  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
      (101.79107615, 26.05004621, 0.99601156)>
  >>> new_hgs_coord.transform_to('icrs')
  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
      (101.79107615, 26.05004621, 0.99601156)>
```
